web/federate.go

```diff
@@ -44,6 +44,10 @@ var (
 	})
 )
 
+func registerFederationMetrics(r prometheus.Registerer) {
```

Comment:
Why do we need a function?

Reply:
so the metric registration is in the same file as the metrics themselves so it is less likely a register is forgotten when adding metrics

do you prefer it differently?